core/components/analytics/provider.tsx

```diff
@@ -1,30 +1,44 @@
 'use client';
 
-import { PropsWithChildren } from 'react';
+import { useConsentManager } from '@c15t/nextjs';
+import { PropsWithChildren, useEffect, useRef } from 'react';
 
 import { FragmentOf } from '~/client/graphql';
 import { Analytics } from '~/lib/analytics';
 import { GoogleAnalyticsProvider } from '~/lib/analytics/providers/google-analytics';
 import { AnalyticsProvider as AnalyticsProviderLib } from '~/lib/analytics/react';
+import { getConsentCookie } from '~/lib/consent-manager/cookies/client';
 
 import { WebAnalyticsFragment } from './fragment';
 
 interface Props {
   channelId: number;
+  isCookieConsentEnabled: boolean;
   settings?: FragmentOf<typeof WebAnalyticsFragment> | null;
 }
 
-const getAnalytics = (
-  channelId: number,
-  settings?: FragmentOf<typeof WebAnalyticsFragment> | null,
-) => {
+const getConsent = () => {
+  const consentCookie = getConsentCookie();
+
+  if (!consentCookie) {
+    return null;
+  }
+
+  return {
+    functionality: consentCookie.preferences.functionality ?? false,
+    marketing: consentCookie.preferences.marketing ?? false,
+    measurement: consentCookie.preferences.measurement ?? false,
+    necessary: consentCookie.preferences.necessary ?? false,
+  };
+};
```

Comment:
Is the only reason we pass this a function is to prevent from defining this functionality on every provider?